lib/AssemblyUtil/FastaToAssembly.py

```diff
@@ -473,6 +482,18 @@ def _validate_mass_params(self, params):
                 raise ValueError(f"Missing {_ASSEMBLY_NAME} field in {_INPUTS} entry #{i}")
         self._get_int(params.get(_MCL), f"If provided, {_MCL}", minimum=2)
 
+        # validate object_metadata is a mapping if provided
+        obj_metas = [i.get(_OBJ_META) for i in inputs]
+        for i, obj_meta in enumerate(obj_metas, start=1):
+            if obj_meta is not None and type(obj_meta) != dict:
+                raise ValueError(f"{_OBJ_META} must be a mapping if provided for entry #{i}")
+            # validate object_metadata keys and values are strings
+            if obj_meta:
```

Comment:
Maybe we don't need if check here?

Reply:
I think we need that since obj_meta can be `None`.

Reply:
What you could do is just do the None check in the conditional and then do the type and key value checks inside that block

Reply:
👍

Reply:
:+1:

Reply:
Will AssemblyUtil.spec do the type check? If not and let's say obj_meta is None, then it will pass this check silently?

Reply:
> Will AssemblyUtil.spec do the type check?

No, the python SDK modules only do type checking for the top level variables. The Java SDK modules type check everything since Java is statically typed

> let's say obj_meta is None, then it will pass this check silently?

It will, but None is a valid input so that's ok

Reply:
👍